src/rest/queue.ts

```diff
@@ -0,0 +1,185 @@
+import { cache } from "./cache.ts";
+import { createRequestBody, processRequestHeaders } from "./request.ts";
+import { HttpResponseCode } from "./types/mod.ts";
+
+/** If the queue is not already processing, this will start processing the queue. */
+export function startQueue() {
+  // IF ALREADY PROCESSING CANCEL
+  if (cache.processingQueue) return;
+  // MARK AS PROCESSING
+  cache.processingQueue = true;
+  processQueue();
+}
+
+/** Processes the queue by looping over each path separately until the queues are empty. */
+export async function processQueue() {
+  while (cache.processingQueue) {
+    // FOR EVERY PATH WE WILL START ITS OWN LOOP.
+    cache.pathQueues.forEach(async (queue) => {
+      // EACH PATH IS UNIQUE LIMITER
+      while (queue.length) {
+        // IF THE BOT IS GLOBALLY RATELIMITED TRY AGAIN
+        if (!cache.globallyRateLimited) continue;
+        // SELECT THE FIRST ITEM FROM THIS QUEUE
+        const [queuedRequest] = queue;
+        // IF THIS DOESNT HAVE ANY ITEMS JUST CANCEL, THE CLEANER WILL REMOVE IT.
+        if (!queuedRequest) return;
+
+        // IF THIS URL IS STILL RATE LIMITED, TRY AGAIN
+        const urlResetIn = checkRateLimits(queuedRequest.payload.url);
+        if (urlResetIn) continue;
+
+        // IF A BUCKET EXISTS, CHECK THE BUCKET'S RATE LIMITS
+        const bucketResetIn = queuedRequest.payload.bucketID
+          ? checkRateLimits(queuedRequest.payload.bucketID)
+          : false;
+        // THIS BUCKET IS STILL RATELIMITED, RE-ADD TO QUEUE
+        if (bucketResetIn) continue;
+
+        // EXECUTE THE REQUEST
```

Comment:
Why every comment in caps? lol

Reply:
Cause i was experimenting with different comment styles and turns out while i was coding i actually liked the caps style. It makes it much more understandable. Code is much easierto readforme